core/state/journal.go

```diff
@@ -154,13 +175,18 @@ type (
 
 func (ch createObjectChange) revert(s *StateDB) {
 	delete(s.stateObjects, *ch.account)
-	delete(s.stateObjectsDirty, *ch.account)
```

Comment:
It's totally wrong to modify the Dirty map, which is the marker for state commit and is only modified at the transaction boundary. An internal state revert should never touch it.